### ISSUE TITLE
New opt: A.sum() - A.reshape(...).sum() -> 0

### ISSUE THREAD

**Issue by nouiz** [Theano/Theano#1347]
We need to merge the sum of variable that are reshape of the same variable.

**Comment by lamblin** [comment]
Indeed, that would be the first step: CAReduce(...)(A.reshape(), axes=None) -> CAReduce(...)(A).
I checked, and it does not seem to be there.
